client/dashboard/app/root/index.tsx

```diff
@@ -126,6 +127,8 @@ function Root() {
 			) }
 		</div>
 	);
+
+	return <BreadcrumbsProvider>{ root }</BreadcrumbsProvider>;
```

Comment:
Something called "BreadcrumbProvider" wrapping the whole app feels dubious.

Also, using a provider at this level can lead to performance issues due to full component tree re-renders. I'm not saying it shouldn't be here, but we should consider ways to avoid providing the context at this level.